lib/index.js

```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
+
+  /**
+   * Get the zoom level the map will move to when there is no bounding box on the selected result
+   * @returns {Number} the map zoom
+   */
+  getZoom: function(){
+    return this.options.zoom;
+  },
+
+  /**
+   * Set the zoom level 
+   * @param {Number} zoom On geocoded result what zoom level should the map animate to when a `bbox` isn't found in the response. If a `bbox` is found the map will fit to the `bbox`.
+   * @returns {MapboxGeocoder} this
+   */
+  setZoom: function(zoom){
+    this.options.zoom = zoom;
+    return this;
+  },
+
+  /**
+   * Get the parameters used to fly to the selected response, if any
+   * @returns {MapboxGeocoder} this
+   */
+  getFlyTo: function(){
+    return this.options.flyTo;
+  },
+
+  /**
+   * Sets the flyTo options
+   * @param {Object|Boolean} flyTo  If false, animating the map to a selected result is disabled. If true, animating the map will use the default animation parameters. If an object, the object will be passed to the flyTo map method to specify a custom animation. 
+   */
+  setFlyTo: function(flyTo){
+    this.options.flyTo = flyTo;
+    return this;
+  },
+
+  /**
+   * Get the value of the placeholder string
+   * @returns {String} The input element's placeholder value
+   */
+  getPlaceholder: function(){
+    return this.options.placeholder;
+  },
+
+  /**
+   * Set the value of the input element's placeholder
+   * @param {String} placeholder the text to use as the input element's placeholder
+   * @returns {MapboxGeocoder} this
+   */
+  setPlaceholder: function(placeholder){
+    this.placeholder = (placeholder) ? placeholder : this._getPlaceholderText();
+    this._inputEl.placeholder = this.placeholder;
+    return this
+  },
+
+  /**
+   * Gets the bounding box used by the plugin
+   * @returns {Array<Number>} the bounding box, if any
+   */
+  getBbox: function(){
+    return this.options.bbox;
+  },
+
+  /**
+   * Sets the bounding box to limit search results to
+   * @param {Array<Number>} bbox a bounding box given as an array in the format [minX, minY, maxX, maxY].
+   * @returns {MapboxGeocoder} this
+   */
+  setBbox: function(bbox){
+    this.options.bbox = bbox;
+    return this;
+  },
+
+  /**
+   * Get a list of the countries to limit search results to 
+   * @returns {String} a comma separated list of countries to limit to, if any
+   */
+  getCountries: function(){
+    return this.options.countries;    
+  },
+
+  /**
+   * Set the countries to limit search results to
+   * @param {String} countries a comma separated list of countries to limit to 
+   * @returns {MapboxGeocoder} this
+   */
+  setCountries: function(countries){
+    this.options.countries = countries;
+    return this;
+  },
+
+  /**
+   * Get a list of the types to limit search results to 
+   * @returns {String} a comma separated list of types to limit to 
+   */
+  getTypes: function(){
+    return this.options.types;    
+  },
+
+  /**
+   * Set the countries to limit search results to
```

Comment:
Should this be types?

```suggestion
   * Set the types to limit search results to
```